src/connection/tcp.c

```diff
@@ -267,7 +268,7 @@ css_hostname_to_ip (const char *host, unsigned char *ip_addr)
       struct hostent hent;
       struct hostent_data ht_data;
 
-      if (gethostbyname_r (host, &hent, &ht_data) == -1)
+      if (gethostbyname_r_uhost (host, &hent, &ht_data) == -1)
```

Comment:
When HAVE_GETHOSTBYNAME_R_HOSTENT_DATA is defined, the gethostbyname_r_uhost() function returns 0 or EFAULT.
It seems that the gethostbyname_r_uhost() error code needs to be changed.

Reply:
Thank you. I will check it again.